zebra-chain/src/primitives/zcash_primitives.rs

```diff
@@ -15,6 +15,131 @@ use crate::{
     transparent::{self, Script},
 };
 
+// Used by boilerplate code below.
+
+#[derive(Clone, Debug)]
+struct TransparentAuth<'a> {
+    all_prev_outputs: &'a [transparent::Output],
+}
+
+impl zcash_primitives::transaction::components::transparent::Authorization for TransparentAuth<'_> {
+    type ScriptSig = zcash_primitives::legacy::Script;
+}
+
+// In this block we convert our Output to a librustzcash to TxOut.
+// (We could do the serialize/deserialize route but it's simple enough to convert manually)
+impl zcash_primitives::transaction::sighash::TransparentAuthorizingContext for TransparentAuth<'_> {
+    fn input_amounts(&self) -> Vec<zcash_primitives::transaction::components::amount::Amount> {
+        self.all_prev_outputs
+            .iter()
+            .map(|prevout| {
+                zcash_primitives::transaction::components::amount::Amount::from_nonnegative_i64_le_bytes(
+                    prevout.value.to_bytes(),
+                ).expect("will not fail since it was previously validated")
+            })
+            .collect()
+    }
+
+    fn input_scriptpubkeys(&self) -> Vec<zcash_primitives::legacy::Script> {
+        self.all_prev_outputs
+            .iter()
+            .map(|prevout| {
+                zcash_primitives::legacy::Script(prevout.lock_script.as_raw_bytes().into())
+            })
+            .collect()
+    }
+}
+
+// Boilerplate mostly copied from `zcash/src/rust/src/transaction_ffi.rs` which is required
+// to compute sighash.
+// TODO: remove/change if they improve the API to not require this.
+
+struct MapTransparent<'a> {
+    auth: TransparentAuth<'a>,
+}
+
+impl<'a>
+    zcash_primitives::transaction::components::transparent::MapAuth<
+        zcash_primitives::transaction::components::transparent::Authorized,
+        TransparentAuth<'a>,
+    > for MapTransparent<'a>
+{
+    fn map_script_sig(
+        &self,
+        s: <zcash_primitives::transaction::components::transparent::Authorized as zcash_primitives::transaction::components::transparent::Authorization>::ScriptSig,
```

Comment:
There is a closing bracket hidden near the end :laughing: Hopefully it's clearer with the shorter types.
If you referring to the `as` keyword... that's good question, I don't know either :sweat_smile: (this code was copied was mentioned above in a comment). It's surprisingly hard to find information about this

Reply:
It's a trait path disambiguation expression:
https://doc.rust-lang.org/reference/paths.html#qualified-paths

It's usually used to disambiguate method calls:
https://doc.rust-lang.org/reference/expressions/call-expr.html#disambiguating-function-calls

It's hard to find in the Rust reference, because it's not listed under the `as` operator.

Reply:
Today I learned, thanks @teor2345!

Reply:
Thanks for the explanatory comments. I had a good laugh when I realized the closing bracket is actually there. I'll use my editor to find it if this happens again.